File: Initial Prototype/Client/MyApplicationProject/MyApplication/src/main/java/com/example/myapplication/domain_objects/ChatMessage.java
package com.example.myapplication.domain_objects;

/**
 * Represents an instant chat message object.
 */
public class ChatMessage {

    private int ChatMessageId;
    private int SenderId;

    public int getChatMessageId() {
        return ChatMessageId;
    }

    public int getSenderId() {
        return SenderId;
    }

    public String getMessageBody() {
        return MessageBody;
    }

    public int getRecipientId() {
        return RecipientId;
    }

    public Boolean getRead() {
        return Read;
    }

    public String getRecipientUserName() {
        return RecipientUserName;
    }

    public String getSenderUserName() {
        return SenderUserName;
    }

    public String getSentOnDate() {
        return SentOnDate;
    }

    private int RecipientId;
    private String MessageBody;
    private String SentOnDate;
    private Boolean Read;
    private String RecipientUserName;
    private String SenderUserName;

    public ChatMessage(int senderId, int recipientId, String messageBody, String sentOnDate, Boolean read, String recipientUserName, String senderUserName)
    {
        this.SenderId = senderId;
        this.RecipientId = recipientId;
        this.MessageBody = messageBody;
        this.SentOnDate = sentOnDate;
        this.Read = read;
        this.RecipientUserName = recipientUserName;
        this.SenderUserName = senderUserName;
    }
}

File: Project/WebService/DomainObjects/Domains/ChatMessage.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="ChatMessage.cs" company="">
//   
// </copyright>
// <summary>
//   The chat message.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace DomainObjects.Domains
{
    using System;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Runtime.Serialization;

    /// <summary>
    /// The chat message.
    /// </summary>
    [DataContract]
    public class ChatMessage
    {
        /// <summary>
        /// Gets or sets the chat message id.
        /// </summary>
        [DataMember]
        [ConcurrencyCheck]
        public int ChatMessageId { get; set; }

        /// <summary>
        /// Gets or sets the sender id.
        /// </summary>
        [DataMember]
        public int SenderId { get; set; }

        /// <summary>
        /// Gets or sets the recipient id.
        /// </summary>
        [DataMember]
        public int RecipientId { get; set; }

        /// <summary>
        /// Gets or sets the message body.
        /// </summary>
        [DataMember]
        [Required(ErrorMessage = "Message is required.")]
        [StringLength(1, MinimumLength = 250,
            ErrorMessage = "Message must be smaller than 250 characters.")]
        public string MessageBody { get; set; }

        /// <summary>
        /// Gets or sets the sent on date.
        /// </summary>
        [DataMember]
        [DataType(DataType.Date)]
        [Column(TypeName = "DateTime2")]
        public DateTime SentOnDate { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether read.
        /// </summary>
        [DataMember]
        public bool Read { get; set; }

        /// <summary>
        /// Gets or sets the recipient user name.
        /// </summary>
        [DataMember]
        public string RecipientUserName { get; set; }

        /// <summary>
        /// Gets or sets the sender user name.
        /// </summary>
        [DataMember]
        public string SenderUserName { get; set; }
    }
}
